sys-apps/ndctl/ndctl-65.ebuild

```diff
@@ -0,0 +1,54 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools bash-completion-r1 linux-info systemd
+
+DESCRIPTION="Utility library for managing the libnvdimm sub-system in the Linux kernel"
+
+HOMEPAGE="http://pmem.io/"
+SRC_URI="https://github.com/pmem/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="LGPL-2.1"
+
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+# All the tests, like the rest of the nvdimm subsystem, require root to run
+# Futhermore, the tests require special testing kernel modules
+RESTRICT="test"
+
+RDEPEND="dev-libs/json-c
+	sys-apps/keyutils
+	sys-apps/kmod
+	virtual/udev"
+
+DEPEND="${RDEPEND}
+	app-text/asciidoc
+	app-text/xmlto"
+
+CONFIG_CHECK="~BLK_DEV_RAM_DAX ~FS_DAX ~X86_PMEM_LEGACY ~LIBNVDIMM ~BLK_DEV_PMEM \
+~ARCH_HAS_PMEM_API ~TRANSPARENT_HUGEPAGE ~MEMORY_HOTPLUG ~MEMORY_HOTREMOVE \
+~ZONE_DEVICE ~FS_DAX_PMD"
+
+src_prepare() {
+	default
+	test -f "${S}/version" || echo "${PV}" > "${S}/version"
+	"${S}"/git-version-gen
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+	  --disable-asciidoctor \
+	  --without-bash \
+	  --without-systemd \
+	  --with-keyutils
```

Comment:
Why are none of these on useflags?

Reply:
For bash and systemd: All the checks to is detect if its there and if so install a bash-completion file and systemd unit respectively. The ebuild unconditionally installs those 2 files.
For asciidoctor: This package needs either asciidoc or asciidoctor to build the man pages, and as asciidoc seems to be the preferred package since it doesn't pull in Ruby
For keyutils: This one could be gated behind a USE flag, but its a small dependency, and doesn't pull anything outside the system set, and a great deal of functionality is gained, I'm not sure if a USE flag would really be useful or not.